pankaj/assignment1/goclient/pod.go

```diff
@@ -0,0 +1,92 @@
+package goclient
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var namespace = "pankaj"
+var podName = "demo-pod"
+
+//PodOperations - perform CURD operations on pod object using client-go library
+func PodOperations() {
+	clientset := KubeConfig()
+	podsClient := clientset.CoreV1().Pods(namespace)
+
+	fileBytes, err := ioutil.ReadFile("configfile/pod.yaml")
+	if err != nil {
+		panic(err.Error())
+	}
+
+	var podSpec apiv1.Pod
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(string(fileBytes))), 1024)
+
+	if err := dec.Decode(&podSpec); err != nil {
+		panic(err)
+	}
+
+	// Create Pod
+	fmt.Println("Pod Create operation started ........")
+	pods, err := podsClient.Create(context.TODO(), &podSpec, metav1.CreateOptions{})
+	if err != nil {
+		panic(err)
+	}
+	time.Sleep(15 * time.Second) //wait to create pod
+	fmt.Printf("Created pod %q.\n", pods.GetObjectMeta().GetName())
+	fmt.Println("Pod Create operation completed ........")
+
+	// Get Pod
+	fmt.Println("Pod Get operation started ........")
+	pods, getErr := podsClient.Get(context.TODO(), podName, metav1.GetOptions{})
+	if getErr != nil {
+
+		panic(fmt.Errorf("Failed to get latest version of Pod: %v", getErr))
```

Comment:
not a good practice to use panic everywhere. Try handling error here and panic in the main func nstead